### PR TITLE
Make style more generic

### PR DESCRIPTION
**Pull request by timunie**
- use pseudo-classes to flip vertically for example
- use nested style to target inner control template. This can be also used to set Stroke etc

### CODE REVIEW

File: src/IconPacks.Avalonia.BoxIcons/PackIconBoxIcons.xaml [hunk 1]
```diff
@@ -4,30 +4,31 @@
                     x:ClassModifier="internal">
 
     <Design.PreviewWith>
-        <iconPacks:PackIconBoxIcons Kind="RegularAtom" />
+        <iconPacks:PackIconBoxIcons Kind="RegularClinic" />
     </Design.PreviewWith>
 
-    <ControlTemplate x:Key="IconPacks.Avalonia.PackIconBoxIcons.Template" TargetType="iconPacks:PackIconBoxIcons">
+    <ControlTemplate x:Key="IconPacks.Avalonia.Template" TargetType="iconPacks:PackIconControlBase">
```
Comment (punker76) [inline src/IconPacks.Avalonia.BoxIcons/PackIconBoxIcons.xaml]: Why removing the PackIconBoxIcons specific type and key name (here and below for the ControlTheme)?

Comment (timunie) [reply src/IconPacks.Avalonia.BoxIcons/PackIconBoxIcons.xaml]: I think it's not needed as this is now more or less one-fits-all unless you see anything that is not covered with pseudo-classses

to add:
- :is-filled
- :is-sroked

Comment (punker76) [reply src/IconPacks.Avalonia.BoxIcons/PackIconBoxIcons.xaml]: Oh, yeah sorry, I think you touched the wrong class, PackIconBoxIcons is realy only for BoxIcons. The [PackIconControl](https://github.com/punker76/MahApps.Avalonia.IconPacks.Testbed/blob/8c87422553d53316f4ac1306af7a4cd8a27aa52[c/src/IconPacks.Avalonia/PackIconControl.cs#L14) is the one for all and with the [unspecific enum Kind](](https://github.com/punker76/MahApps.Avalonia.IconPacks.Testbed/blob/8c87422553d53316f4ac1306af7a4cd8a27aa52c/src/IconPacks.Avalonia/PackIconControl.cs#L22)

Comment (timunie) [reply src/IconPacks.Avalonia.BoxIcons/PackIconBoxIcons.xaml]: 🙈 you are right, sure

**Comment by timunie** [comment]
I updated the proposal. I now even like the idea more. If you also like the idea, please add the :hacktoberfest: label. 

I can work on another PR where I do the same for `:filled` and `:outlined`-pseudoclasses. This would then cover almost all of the ControlTemplate-Triggers you had. Let me know your comments.

**Comment by punker76** [comment]
your changes have been merged, thanks for your contribution 👍

**Pull request merged**